test/ts/unit/simple_interest_terms_contract.ts

```diff
@@ -0,0 +1,237 @@
+          await expect(termsContract.repaymentRouter.callAsync()).to.eventually.equal(router.address);
+        });
+        it("points to the token contract passed in through the constructor", async () => {
+          await expect(termsContract.repaymentToken.callAsync()).to.eventually.equal(mockToken.address);
+        });
+    });
+
+    describe("#registerRepayment", () => {
+        describe("agent who is not RepaymentRouter calls registerRepayment", () => {
+            it("should throw", async () => {
+              await expect(termsContract.registerRepayment.sendTransactionAsync(
+                ARBITRARY_AGREEMENT_ID,
+                PAYER,
+                BENEFICIARY,
+                Units.ether(1),
+                mockToken.address,
+                { from: ATTACKER } // the message is coming from an attacker.
+              )).to.eventually.be.rejectedWith(REVERT_ERROR);
+            });
+        });
+
+        describe("RepaymentRouter calls registerRepayment", () => {
+
+            describe("...with a different `tokenAddress` than expected by the terms contract", () => {
+
+                before(async () => {
+                    await router.repay.sendTransactionAsync(
+                        ARBITRARY_AGREEMENT_ID,
+                        Units.ether(1),
+                        NULL_ADDRESS // this is not the address it's expecting.
+                    );
+                });
+
+                it("should not record the repayment in any capacity", async () => {
+                  await expect(termsContract.getValueRepaidToDate.callAsync(
+                    ARBITRARY_AGREEMENT_ID)).to.eventually.equal(Units.ether(0));
+                });
+            });
+
+            describe("...with terms contract's expected `tokenAddress`", () => {
+
+                before(async () => {
+                    await router.repay.sendTransactionAsync(
+                        ARBITRARY_AGREEMENT_ID,
+                        Units.ether(1),
+                        mockToken.address // this is the address it's expecting.
+                    );
+                });
+
+                it("should record the repayment", async () => {
+                  await expect(termsContract.getValueRepaidToDate.callAsync(
+                    ARBITRARY_AGREEMENT_ID)).to.eventually.equal(Units.ether(1));
+                });
+
+            });
+        });
+    });
+
+    describe("#getValueRepaidToDate", () => {
+
+        describe("non-existent debt agreement", () => {
+            /*
+            Question(kayvon): as I understand it, the contract wouldn't throw,
+            but just return 0. Do we want to change this implementation?
+            */
+            it("should throw");
+        });
+
+        describe("extant debt agreement", () => {
+
+          describe("at time N, debtor has made an initial payment of 1 ether", () => {
+
+              before(async () => {
+                  await router.repay.sendTransactionAsync(
```

Comment:
@saturnial You'll want to mock the `getBeneficiary` and `getTermsContract` functions of the debt registry to return the parameters you want for `ARBITRARY_AGREEMENT_ID`.

This is because `repay` has two different invariants it enforces on repayment operations:

1. The debt agreement has to exist (see [here](https://github.com/dharmaprotocol/charta/blob/78bacda84e284084f331c65100a09a06c6be892a/contracts/RepaymentRouter.sol#L85))
2. The terms contract has to return `true` to `registerRepayment` (see [here](https://github.com/dharmaprotocol/charta/blob/78bacda84e284084f331c65100a09a06c6be892a/contracts/RepaymentRouter.sol#L100))

Both of these require you to mock, respectively, `getBeneficiary` and `getTermsContract` for `ARBITRARY_AGREEMENT_ID`.

Reply:
something like this?

before(async () => {
              await mockRegistry.mockGetBeneficiaryReturnValueFor.sendTransactionAsync(
                  ARBITRARY_AGREEMENT_ID,
                  BENEFICIARY,
              );

              await mockRegistry.mockGetTermsReturnValueFor.sendTransactionAsync(
                  ARBITRARY_AGREEMENT_ID,
                  termsContract.address,
                  TERMS_CONTRACT_PARAMETERS,
              );

              await mockRegistry.mockGetTermsContractReturnValueFor.sendTransactionAsync(
                  ARBITRARY_AGREEMENT_ID,
                  termsContract.address,
              );
          });

Reply:
Yup, LGTM :)

Reply:
w00t